streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -160,6 +163,11 @@ public void handleAssignment(final Map<TaskId, Set<TopicPartition>> activeTasks,
             logPrefix
         );
 
+        // initialize the created tasks
```

Comment:
See other comment above: we move the initialization logic right after creation so that within the iteration we only check for completion.